### PR TITLE
Clear selected properties on deselecting all

### PR DESCRIPTION
**Pull request by malberts**
Refs https://github.com/ProfessionalWiki/WikibaseExport/pull/35#issuecomment-1311695364

I couldn't immediately find an OOUI method that does the same. There might not actually be a lot of OOUI use cases for clearing everything like this, so maybe I'm not totally missing it.

### CODE REVIEW

File: resources/ext.wikibase.export.js [hunk 1]
```diff
@@ -132,6 +132,10 @@ $( function () {
 					widget.statements.toggle( false );
 					widget.statements.setValue( widget.statements.menu.getItems() );
 				} else {
+					// TODO: is there an OOUI way to clear everything?
+					widget.statements.getValue().forEach( function ( statement ) {
+						widget.statements.removeTagByData( statement );
+					} );
```
Comment (JeroenDeDauw) [inline resources/ext.wikibase.export.js:138]: My JS is a bit rusty so I'm not sure you can actually pass a reference to the function directly into the higher-order forEach. Nowadays you can even do this in PHP (8.1+)

Comment (JeroenDeDauw) [reply resources/ext.wikibase.export.js:138]: I mean

```js
widget.statements.getValue().forEach( widget.statements.removeTagByData );
```

Comment (malberts) [reply resources/ext.wikibase.export.js]: Doesn't look like it. The closest I can get is
```js
widget.statements.getValue().forEach( statement => widget.statements.removeTagByData( statement ) );

```

Comment (JeroenDeDauw) [reply resources/ext.wikibase.export.js]: Let's see how functional my twitter followers end up being https://twitter.com/JeroenDeDauw/status/1591206141928779777 :)

**Pull request merged**